### PR TITLE
Allow using external cluster ID to run logging check

### PR DESCRIPTION
**Pull request by NautiluX**
* Alerts contain external ID as label
* This change will prevent an additional manual lookup step

jira: [OSD-12082](https://issues.redhat.com//browse/OSD-12082)

### CODE REVIEW

File: pkg/utils/ocm.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+// GenerateQuery returns an OCM search query to retrieve all clusters matching an expression (ie- "foo%")
+func GenerateQuery(clusterIdentifier string) string {
+	return strings.TrimSpace(fmt.Sprintf("(id like '%[1]s' or external_id like '%[1]s' or display_name like '%[1]s')", clusterIdentifier))
+}
+
+func CreateConnection() *sdk.Connection {
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		if strings.Contains(err.Error(), "Not logged in, run the") {
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply pkg/utils/ocm.go:68]: I guess it's enough to match the error string expected? (Note this function is just moved and made public)

File: cmd/cluster/loggincheck.go [hunk 2]
```diff
@@ -60,7 +60,24 @@ func (o *loggingCheckOptions) complete(cmd *cobra.Command, args []string) error
...
+	// the user has to be logged in (e.g. 'ocm login')
+	ocmClient := utils.CreateConnection()
+	defer func() {
+		if err := ocmClient.Close(); err != nil {
+			fmt.Printf("Cannot close the ocmClient (possible memory leak): %q", err)
+		}
+	}()
+
+	if len(args) != 1 {
+		return fmt.Errorf("too many arguments. Expected 1 got %d", len(args))
```
Comment (fahlmant) [inline cmd/cluster/loggincheck.go:73]: what if there's too few? ;)

**Review by fahlmant — APPROVED** [review]
/approve
/lgtm

/hold
Feel free to unhold when you want

File: cmd/cluster/loggincheck.go (hunk 2 above)
Comment (NautiluX) [reply cmd/cluster/loggincheck.go:73]: hah, good catch!

**Review by fahlmant — APPROVED** [review]
/lgtm

**Pull request merged**